tuning up custom jobs

Jolly-66 opened the pull request:
thanks melbert and timber for giving me an initial nightmare initially

- Restored the heirlooms of the 3 jobs, BO has a new one, donuts (also modified existing heirlooms for RD and scientist; they're pens since they start with a clipboard)
- Modified access for QM and scientist. Toxi & Xenobio both have the same low pop access as scientist. Scientist cannot access these rooms outside of lowpop. QM can access sci on low pop to yell at the lazy scientist as to why theres no mining tech (or run to the bridge and do it them damn selves). BO is also less of a HoP backbone and more specialized. May give them more sec access in the future (low pop they can still access SOME service areas). 
- QM starts with a baton and has a silver ID, oops!
- Tweaked Lavaland ruins
- Untainted main cards.dmi to favor the modular cards.dmi (and to prevent any future merge conflicts)

Jolly-66 commented:
fuck off checks

sqnztb commented on code/modules/jobs/job_types/research_director.dm:
```diff
@@ -23,7 +23,7 @@
 	bounty_types = CIV_JOB_SCI
 
-	family_heirlooms = list(/obj/item/toy/plush/slimeplushie)
+	family_heirlooms = list(/obj/item/pen) //NON MODULAR CHANGE
```
I don't particularly agree with this, since you can just slot the pen into your PDA and never have to worry about losing your heirloom.

sqnztb commented on code/modules/jobs/job_types/scientist.dm:
```diff
@@ -18,7 +18,7 @@
 	departments = DEPARTMENT_SCIENCE
 
-	family_heirlooms = list(/obj/item/toy/plush/slimeplushie)
+	family_heirlooms = list(/obj/item/pen) //NON MODULAR CHANGE
```
Same as the RD heirloom.

Jolly-66 replied:
i don't wanna make it a clipboard, they start with it
slime plushie works better with the xenobio job
would a beaker or something make more sense? or even a book on something?

sqnztb commented on code/datums/id_trim/jobs.dm:
```diff
@@ -343,8 +343,10 @@
-	access = list(ACCESS_MAINT_TUNNELS, ACCESS_MAILSORTING, ACCESS_CARGO, ACCESS_QM, ACCESS_MINING, ACCESS_MECH_MINING, ACCESS_MINING_STATION, ACCESS_MINERAL_STOREROOM, ACCESS_VAULT, ACCESS_AUX_BASE, ACCESS_HEADS, ACCESS_SEC_DOORS, ACCESS_SECURITY) //NON-MODULAR CHANGES
-	minimal_access = list(ACCESS_MAINT_TUNNELS, ACCESS_MAILSORTING, ACCESS_CARGO, ACCESS_QM, ACCESS_MINING, ACCESS_MECH_MINING, ACCESS_MINING_STATION, ACCESS_MINERAL_STOREROOM, ACCESS_VAULT, ACCESS_AUX_BASE, ACCESS_HEADS, ACCESS_SEC_DOORS, ACCESS_SECURITY) //NON-MODULAR CHANGES
+	access = list(ACCESS_MAINT_TUNNELS, ACCESS_MAILSORTING, ACCESS_CARGO, ACCESS_QM, ACCESS_MINING, ACCESS_MECH_MINING, ACCESS_MINING_STATION, 
+					ACCESS_MINERAL_STOREROOM, ACCESS_VAULT, ACCESS_AUX_BASE, ACCESS_HEADS, ACCESS_SEC_DOORS, ACCESS_SECURITY, ACCESS_CHANGE_IDS, ACCESS_ENGINE, ACCESS_RND, ACCESS_RESEARCH) //NON-MODULAR CHANGES
```
I haven't reviewed the code on the new ID system. Does `ACCESS_CHANGE_IDS` give the full ID card modification? So QM has just as much power as the HOP now on lowpop? Also `ACCESS_ENGINE`?

Jolly-66 replied:
i can ask timber about it, i doubt they can modify anything outside of their access, considering heads have the same access parameter

Jolly-66 replied:
access engine was suggested by @rellyreri

Jolly-66 replied:
ok yeah that is a bad idea (change_ids)
i'll edit it out, and if i can/want to, make a subtype for qm later, may not be needed at all but it'll be an idea i'll keep in the back of my head

sqnztb commented on config/game_options.txt:
```diff
@@ -516,13 +516,13 @@ BOMBCAP 20
 ## spawn, while the converse is true. Alter this number to affect the amount
 ## of ruins.
-LAVALAND_BUDGET 45
+LAVALAND_BUDGET 30
```
What's the reasoning on cutting lavaland+space ruin budget by a third? Surely we don't have that much gamer-ing going on currently.

Jolly-66 replied:
lavaland is mostly due to cutting down on certain ruins
instead of having all of them spawn at once, limiting them sounds better
i don't know the internal values for the ruins, so if someone could bring them to me i can reconsider greater

MrMelbert commented on code/datums/id_trim/jobs.dm:
```diff
@@ -383,7 +385,7 @@
 	access = list(ACCESS_ROBOTICS, ACCESS_RND, ACCESS_TOXINS, ACCESS_TOXINS_STORAGE, ACCESS_RESEARCH, ACCESS_XENOBIOLOGY,
 					ACCESS_MECH_SCIENCE, ACCESS_MINERAL_STOREROOM, ACCESS_TECH_STORAGE, ACCESS_GENETICS, ACCESS_AUX_BASE)
-	minimal_access = list(ACCESS_RND, ACCESS_TOXINS, ACCESS_TOXINS_STORAGE, ACCESS_RESEARCH, ACCESS_XENOBIOLOGY, ACCESS_MECH_SCIENCE,
+	minimal_access = list(ACCESS_RND, ACCESS_RESEARCH, ACCESS_MECH_SCIENCE,
 							ACCESS_MINERAL_STOREROOM, ACCESS_AUX_BASE)
```
missing comment for non-module change

sqnztb commented on config/lavaruinblacklist.txt:
```diff
@@ -32,10 +32,12 @@ _maps/RandomRuins/LavaRuins/lavaland_surface_sloth.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_gaia.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_elephant_graveyard.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_library.dmm
-#_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
```
Cult altar I get, but why blacklist the healing fountain, the puzzle cube, and the knight armor ruin?

MrMelbert replied:
`_cube.dmm` isn't the puzzle cube, it's the wishgranter cube

sqnztb replied:
Ah, righto then.

MrMelbert commented on code/datums/mapgen/Cavegens/LavalandGenerator.dm:
```diff
@@ -5,7 +5,7 @@
 	mob_spawn_list = list(/mob/living/simple_animal/hostile/asteroid/goliath/beast/random = 50, /obj/structure/spawner/lavaland/goliath = 3, \
 		/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/random = 40, /obj/structure/spawner/lavaland = 2, \
-		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/random = 30, /obj/structure/spawner/lavaland/legion = 3, \
+		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/random = 30, /obj/structure/spawner/lavaland/legion = 2, \ //NON-MODULAR CHANGE
```
This doesn't change too terribly much in the grand scheme of things, but I guess

sqnztb commented on jollystation_modules/code/modules/jobs/job_types/bridge_officer.dm:
```diff
@@ -24,6 +24,8 @@
 
 	display_order = JOB_DISPLAY_ORDER_BRIDGE_OFFICER
 
+	family_heirlooms = list(/obj/item/book/manual/wiki/security_space_law, /obj/item/storage/fancy)
```
What happens when you fold the donut box up into cardboard? Does this just kill your heirloom permanently?

MrMelbert replied:
yep

sqnztb replied:
Also I just checked, `/obj/item/storage/fancy` is one of the item basetypes that has an ERROR sprite, if we're doing a donut box heirloom, should be `/obj/item/storage/fancy/donut_box` shouldn't it?

MrMelbert commented on config/lavaruinblacklist.txt:
```diff
@@ -32,10 +32,12 @@ _maps/RandomRuins/LavaRuins/lavaland_surface_sloth.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_hierophant.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_pizzaparty.dmm
-#_maps/RandomRuins/LavaRuins/lavaland_surface_cultaltar.dmm
+_maps/RandomRuins/LavaRuins/lavaland_surface_cultaltar.dmm
```
This ruin has real cult items but none of it's usable, not really a problem

Jolly-66 replied:
don't the jumpsuits have armor in them? unlikely that someone would put them on, but its a consideration

MrMelbert commented on config/lavaruinblacklist.txt:
```diff
@@ -32,10 +32,12 @@ _maps/RandomRuins/LavaRuins/lavaland_surface_sloth.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_elephant_graveyard.dmm
 #_maps/RandomRuins/LavaRuins/lavaland_surface_library.dmm
-#_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
+_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
```
This ruin has nothing but butterflies and an altar that gives you omnizine at the end, not really a problem

MrMelbert commented on config/lavaruinblacklist.txt:
```diff
@@ -32,10 +32,12 @@ _maps/RandomRuins/LavaRuins/lavaland_surface_sloth.dmm
-#_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
+_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
 _maps/RandomRuins/LavaRuins/lavaland_surface_swarmer_crash.dmm
+_maps/RandomRuins/LavaRuins/lavaland_surface_cube.dmm
```
good

MrMelbert commented on config/lavaruinblacklist.txt:
```diff
@@ -32,10 +32,12 @@ _maps/RandomRuins/LavaRuins/lavaland_surface_sloth.dmm
+_maps/RandomRuins/AnywhereRuins/fountain_hall.dmm
 _maps/RandomRuins/LavaRuins/lavaland_surface_swarmer_crash.dmm
+_maps/RandomRuins/LavaRuins/lavaland_surface_cube.dmm
+_maps/RandomRuins/LavaRuins/lavaland_surface_wizard.dmm
```
This ruin is rarer and contains the disk for knight armor / ruinite, I can't really see this being a problem

Jolly-66 commented:
changes

- bo no longer has a box of error donuts as an heirlooms, its space law
- scientist and rd now share the cytology or beaker as their heirloom
- removed id access from qm, and engine from their innate access (kept it on lowpop for now, may move back to innate if deemed; reasoning was that cargo can work with engineering on projects, blah blah blah) 
- added back that not actually bad lavaland ruin
- added missing non modular comment
